### PR TITLE
Reduce TTL and add a temporary sleep before etcd cluster formation.

### PR DESCRIPTION
**Pull request by davidewatson**

### CODE REVIEW

**Comment by davidewatson** [comment]
This is an experimental attempt to fix #673.

File: ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.etcd.part.jinja2 [hunk 1]
```diff
@@ -19,7 +19,9 @@
     RestartSec=15
```
Comment (mikeln) [inline ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.etcd.part.jinja2]: as long as you are in this area.. .this value should be moved into the `defaults/main.yaml`

File: ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.etcd.part.jinja2 [hunk 2]
```diff
@@ -19,7 +19,9 @@
...
     Restart=always
     KillMode=none
-    ExecStartPre=/usr/bin/mkdir -p /ephemeral/etcd
+    # This sleep is a test.
+    ExecStartPre=/usr/bin/mkdir -p /ephemeral/etcd && \
+     sleep 300
```
Comment (mikeln) [inline ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.etcd.part.jinja2]: Same here... this value should be set in `defaults/main.yaml`

File: ansible/roles/kraken.provider/kraken.provider.aws/templates/kraken.provider.aws.module.etcd.tf.jinja2 [hunk 3]
```diff
@@ -123,7 +123,7 @@ resource "aws_route53_record" "{{node.name}}_vm_A_record_{{node_idx}}" {
   zone_id   = "${var.route53_zone_id}"
   name      = "${aws_instance.{{node.name}}_{{node_idx}}.id}.{{node.name}}.${var.etcds_name}.internal"
   type      = "A"
-  ttl       = "300"
+  ttl       = "30"
```
Comment (mikeln) [inline ansible/roles/kraken.provider/kraken.provider.aws/templates/kraken.provider.aws.module.etcd.tf.jinja2]: same here... should be in `defaults\main.yaml`

File: ansible/roles/kraken.provider/kraken.provider.aws/templates/kraken.provider.aws.module.etcd.tf.jinja2 [hunk 4]
```diff
@@ -133,7 +133,7 @@ resource "aws_route53_record" "{{node.name}}_A_record_all" {
   zone_id   = "${var.route53_zone_id}"
   name      = "{{node.name}}.${var.etcds_name}.internal"
   type      = "A"
-  ttl       = "300"
+  ttl       = "30"
```
Comment (mikeln) [inline ansible/roles/kraken.provider/kraken.provider.aws/templates/kraken.provider.aws.module.etcd.tf.jinja2]: same as above.

**Comment by davidewatson** [comment]
Results from first CI run show that both etcd clusters formed successfully.

> [aws]                     "member 11de4d52b0ce691b is healthy: got healthy result from https://10.0.13.160:2379", 
[aws]                     "member 5aa618160a184b91 is healthy: got healthy result from https://10.0.102.28:2379", 
[aws]                     "member 7683fec16ad2a1ea is healthy: got healthy result from https://10.0.92.185:2379", 
[aws]                     "member ea6f46225f09d17c is healthy: got healthy result from https://10.0.226.86:2379", 
[aws]                     "member f87fbccbf83be0d1 is healthy: got healthy result from https://10.0.218.68:2379", 

> [aws]                     "member 110141e2926058ec is healthy: got healthy result from https://10.0.220.79:2381", 
[aws]                     "member 1b2dc41785c9b03e is healthy: got healthy result from https://10.0.167.153:2381", 
[aws]                     "member 1f7abb9ca04d7294 is healthy: got healthy result from https://10.0.75.158:2381", 
[aws]                     "member 51265a02a6a6c561 is healthy: got healthy result from https://10.0.21.244:2381", 
[aws]                     "member afe1d20844f6434e is healthy: got healthy result from https://10.0.122.165:2381",

**Comment by coffeepac** [comment]
that e2e failure is nothing.  totally ignorable.

**Comment by davidewatson** [comment]
The second CI run failed:

> [aws] FAILED - RETRYING: Wait for etcd cluster to form (1 retries left).
[aws] failed: [localhost] (item=5) => {"attempts": 30, "changed": true, "cmd": ["ssh", "-o", "StrictHostKeyChecking=no", "-o", "UserKnownHostsFile=/dev/null", "-F", "/home/jenkins/workspace/K2_PR-707-VTYZFVLG3VIQJ5YZ4RXVY7QZIKIY6QO2CBAUKVM7YRJOJOT6OWDA/cluster/aws//pr-707-2/ssh_config", "etcd-5", "etcdctl", "--ca-file=/etc/etcd/ssl/client-ca.pem", "--cert-file=/etc/etcd/ssl/client.pem", "--key-file=/etc/etcd/ssl/client-key.pem", "--endpoints=https://127.0.0.1:2379,https://127.0.0.1:4001", "cluster-health"], "delta": "0:00:02.519083", "end": "2017-08-03 20:59:03.742544", "failed": true, "idx": "5", "rc": 4, "start": "2017-08-03 20:59:01.223461", "stderr": "Warning: Permanently added 'ec2-18-220-48-52.us-east-2.compute.amazonaws.com,18.220.48.52' (ECDSA) to the list of known hosts.\r\nError:  client: etcd cluster is unavailable or misconfigured\nerror #0: client: endpoint https://127.0.0.1:4001 exceeded header timeout\nerror #1: client: endpoint https://127.0.0.1:2379 exceeded header timeout", "stderr_lines": ["Warning: Permanently added 'ec2-18-220-48-52.us-east-2.compute.amazonaws.com,18.220.48.52' (ECDSA) to the list of known hosts.", "Error:  client: etcd cluster is unavailable or misconfigured", "error #0: client: endpoint https://127.0.0.1:4001 exceeded header timeout", "error #1: client: endpoint https://127.0.0.1:2379 exceeded header timeout"], "stdout": "cluster may be unhealthy: failed to list members", "stdout_lines": ["cluster may be unhealthy: failed to list members"]}

**Comment by davidewatson** [comment]
Third CI run failed.

> [aws]                     "member 12f02725d8ebcec4 is unreachable: no available published client urls", 
[aws]                     "member 3bf951b0f3e2ec91 is healthy: got healthy result from https://10.0.255.174:2381", 
[aws]                     "member 829c73aa9d0a9500 is healthy: got healthy result from https://10.0.119.54:2381",

**Comment by davidewatson** [comment]
The etcd environment is not being setup correctly.

**Pull request closed without merging**